cdisc_rules_engine/standards/sdtm_standards_context.py

```diff
@@ -597,3 +632,67 @@ def _do_child_merge(
             merge_spec=merge_spec,
         )
         return result_schema.name
+
+    def detect_split_datasets(self, dataset_names: List[str]) -> Dict[str, List[str]]:
+        """
+        Detect split datasets by name.
+        """
+        split_groups = defaultdict(list)
+
+        datasets = [name.lower() for name in dataset_names]
+
+        for dataset in datasets:
+            unsplit_name = self._get_unsplit_name(dataset)
+
+            if unsplit_name != dataset:
+                split_groups[unsplit_name].append(dataset)
+
+        return {k: v for k, v in split_groups.items() if len(v) > 1}
+
+    def _get_unsplit_name(self, dataset_name: str) -> str:
+        """
+        Extract the unsplit (logical) name from a dataset name following
+        SDTMIG v3.4 naming conventions.
+
+        Detection rules from SDTMIG v3.4 Section 4.1.7:
+        - AE -> not split (domain only)
+        - AE1, AE2 -> split (domain + number/letter suffix)
+        - SUPPAE1 -> split (SUPP + domain + digit)
+        - FACM, FAEG -> split Findings About (FA + 2-char parent domain code)
+        - SUPPQS36, SUPPQSPI, SUPPFACM -> split supplemental qualifiers
+        """
+        dataset = dataset_name.lower()
+
+        # Pattern 1: Domain + digit(s) (e.g. AE1, AE2, QS36)
+        match = re.match(r"^([a-z]{2,4})(\d+)$", dataset)
```

Comment:
Loads of these still use the wrong regex. From my reading of the standards, the ID can be any combination of digits/characters